packages/core-typings/src/IUser.ts

```diff
@@ -153,7 +153,9 @@ export interface IRegisterUser extends IUser {
 	username: string;
 	name: string;
 }
+
 export const isRegisterUser = (user: IUser): user is IRegisterUser => user.username !== undefined && user.name !== undefined;
+export const isUserFederated = (user: Partial<IUser>): user is IUser => 'federated' in user && user.federated === true;
```

Comment:
since this is not a type assertion, I'm not sure this should be here.. it is conceptually different from `isRegisterUser` that asserts a new type (`IRegisterUser`) if some validation is `true`.

Reply:
I followed the idea of `isRoomFederated()`, https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/core-typings/src/IRoom.ts#L90. Where do you think it should be?

Reply:
ok, `IRoomFederated()` is fine since it is indeed asserting the type `IRoomFederated` ... you could create a `IUserFederated` here, where the `federated` property is not optional.. that would be a correct usage.